libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt

```diff
@@ -426,6 +439,14 @@ abstract class KotlinCompile @Inject constructor(
     @get:Input
     abstract val useClasspathSnapshot: Property<Boolean>
 
+    @get:Classpath
+    @get:Optional // Set if useClasspathSnapshot == true
+    abstract val classpathSnapshot: ConfigurableFileCollection
+
+    @get:OutputDirectory
+    @get:Optional // Set if useClasspathSnapshot == true
+    abstract val classpathSnapshotDir: DirectoryProperty
+
```

Comment:
Would creating a `@Nested` input make this more elegant?

Reply:
Done